### ISSUE TITLE
Cancel the build for secrets leak commit

### ISSUE THREAD

**Issue by kaleocheng** [Wiredcraft/github-actions#4]
The current build process only check if there is any secret in the code and upload the report but still do the build. means developer can still deploy it to envs, no one will be noticed for the secret leak.

**Comment by guinan34** [comment]
indeed.  i updated the scan strategy. 

1. we need to defined which leak level will to stop the actions fisrt. <img width="895" alt="image" src="https://user-images.githubusercontent.com/16395409/168546692-45eed79d-1050-4eac-8e85-70659460d0a9.png">
2.i can add 2 task to check if there are high level leak , then return an  `exit 1` to stop the actions. 
below is an example:
```
    - name: Exit actions if there were some critical secret leak
       id: leak_check
       run: echo "::set-output name=leak_level::$(grep 'CRITICAL' /tmp/scan_output.json |wc -l)"
     - name: Pass or fail
       run: |
         if [[ ${{steps.leak_check.outputs.leak_level}} == 0 ]]; then exit 0; else echo "here are some CRITICAL secret leak , pls check the actions artifacts . (click Actions , then click your action task , then you can find the Artifacts.)" && exit 1; fi
```
<img width="1869" alt="image" src="https://user-images.githubusercontent.com/16395409/168546223-eb439072-c6f6-4d03-8034-381520569849.png">

how do you think this strategy ?

**Comment by sp3c73r2038** [comment]
how is those levels sorted? 

```
CRITICAL > BLOCKER > MAJOR > MINOR > INFO
```

in that case, anything except INFO might be reported to stop the build.

**Comment by guinan34** [comment]
indeed,  the INFO level will check the comment, looks like the regex not really sensitive.
<img width="491" alt="image" src="https://user-images.githubusercontent.com/16395409/169739515-42162641-2d9f-4d64-8544-8aaf927dfa8d.png">

**Comment by guinan34** [comment]
but the main problem is there are so many mis-report , such as the key is a sensitive key word, but the value is a ansible jinjia2 variable, so this should not be a leak. if we use this tool, we should adjust more rule to adapted our code style.

<img width="1673" alt="image" src="https://user-images.githubusercontent.com/16395409/169741401-dfcee8c9-3e56-4d0d-9cbc-5113c8b6cb36.png">

**Comment by guinan34** [comment]
so the main work is to adjusted the regex to adapted our code sytle.

<img width="480" alt="image" src="https://user-images.githubusercontent.com/16395409/169741955-3be8afef-7a3d-4f75-82fc-2f0c862a4231.png">

**Comment by kaleocheng** [comment]
@guinan34 I'm good at the strategy in here( break for any level > INFO), but seems the tool ( whispers) we are using is kind of hard to implement those,  should we migrate a more model tool e.g. https://github.com/aquasecurity/trivy?

**Comment by guinan34** [comment]
sure, will investigation more new tools .

**Comment by guinan34** [comment]
@kaleocheng  when  i tried to use the Trivy in github-action.  it can't stop the build process even the critically error has been scanned out.  i followed the official doc to add the `exit-code: 1` parameter but didn't work. so i create an issues for this at their github repo . this is the latest update for now . 

https://github.com/aquasecurity/trivy-action/issues/135

**Comment by guinan34** [comment]
https://github.com/Wiredcraft/github-actions/pull/6   @sp3c73r2038 @kaleocheng   
they were fixed the exit-code bug. here is the PR . pls review

**Comment by kaleocheng** [comment]
closing as done
